src/bentoml/bentos.py

```diff
@@ -441,81 +441,62 @@ def serve(
     max_concurrent_streams: int
     | None = Provide[BentoMLContainer.grpc.max_concurrent_streams],
     grpc_protocol_version: str | None = None,
-) -> ServerHandle:
-    from .serve import construct_ssl_args
-    from ._internal.server.server import ServerHandle
-
-    if isinstance(bento, Bento):
-        bento = str(bento.tag)
-    elif isinstance(bento, Tag):
-        bento = str(bento)
-
-    server_type = server_type.lower()
-    if server_type not in ["http", "grpc"]:
-        raise ValueError('Server type must either be "http" or "grpc"')
-
-    ssl_args: dict[str, t.Any] = {
-        "ssl_certfile": ssl_certfile,
-        "ssl_keyfile": ssl_keyfile,
-        "ssl_ca_certs": ssl_ca_certs,
-    }
+) -> Server:
+    logger.warning(
+        "bentoml.serve and bentoml.bentos.serve are deprecated; use bentoml.Server instead."
+    )
+
     if server_type == "http":
-        serve_cmd = "serve-http"
+        from .server import HTTPServer
+
         if host is None:
-            host = BentoMLContainer.http.host.get()
+            host = t.cast(str, BentoMLContainer.http.host.get())
```

Comment:
I think we should use this as the default value in serve as well? so maybe we can change it for `host`, `port`, `api_workers`?

Reply:
This should probably be removed, I made the conscious decision to only support HTTP with `bentoml.serve` for now.

Reply:
Oh, that's not true at all. The rest are provided by injection, I think.

Reply:
I think for all of the inject we can just make it as default, so that users can use the environment variable with Python API.

Reply:
I don't think I understand, what're you proposing?

Reply:
We should make any value that is injectable as the default, so that if users specify these configuration via the environment variable it will work when user use the Python API + envvar?

Reply:
I don't think I understand, isn't that already handled by injection?

Reply:
I think what i mean is 
```python
        host: str = Provide[BentoMLContainer.http.host],
        port: int = Provide[BentoMLContainer.http.port],
```

Reply:
That doesn't work because it's different for gRPC, besides, this is functionally identical.